tests/SonarScanner.Shim.Tests/PathHelperTests.cs

```diff
@@ -0,0 +1,187 @@
+﻿/*
+ * SonarQube Scanner for MSBuild
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarScanner.Shim.Tests
+{
+    [TestClass]
+    public class PathHelperTests
+    {
+        [TestMethod]
+        public void WithTrailingDirectorySeparator_WhenNull_ThrowsArgumentNullException()
+        {
+            // Arrange & Act
+            Action act = () => PathHelper.WithTrailingDirectorySeparator(null);
+
+            // Assert
+            act.ShouldThrow<ArgumentNullException>().And.ParamName.Should().Be("directory");
+        }
+
+        [TestMethod]
+        public void WithTrailingDirectorySeparator_WhenDirectoryEndsWithBackslash_ReturnsSameString()
+        {
+            // Arrange
+            var path = "SomeDirectory\\";
+
+            // Act
+            var result = PathHelper.WithTrailingDirectorySeparator(new DirectoryInfo(path));
+
+            // Assert
+            result.Should().EndWith("\\");
+        }
+
+        [TestMethod]
+        public void WithTrailingDirectorySeparator_WhenDirectoryEndsWithSlash_ReturnsSameString()
```

Comment:
The test name doesn't match the assertion: it's returning a different string.
I don't think we'll be running these tests on a non-Windows machine any time soon, but it might be better to use _Path.DirectorySeparator/AltDirectorySeparator_ in the assertions, just to be on the safe side.

(and thinking about it, maybe running the tests on a non-Windows machine wouldn't be such a bad idea)